pkg/synthetics/types.go

```diff
@@ -502,12 +590,142 @@ func (x *Nr1CatalogTargetedInstallPlanDirective) ImplementsNr1CatalogInstallPlan
+	Repository string `json:"repository,omitempty"`
+}
+
+// SyntheticsAutomatedTestConfigInput - Global automated test config
+type SyntheticsAutomatedTestConfigInput struct {
+	// An identifier for the automated test run
+	BatchName string `json:"batchName,omitempty" yaml:"batchName,omitempty"`
+	// Branch metadata to indicate what triggered the automated test
+	Branch string `json:"branch,omitempty" yaml:"branch,omitempty"`
+	// Commit metadata to indicate what commit triggered the automated test
+	Commit string `json:"commit,omitempty" yaml:"commit,omitempty"`
+	// Metadata for the automated test
+	DeepLink string `json:"deepLink,omitempty" yaml:"deepLink,omitempty"`
+	// Metadata about the platform target the automated test will run against
+	Platform string `json:"platform,omitempty" yaml:"platform,omitempty"`
+	// The unique client identifier for the Synthetics Monitor in New Relic
+	Repository string `json:"repository,omitempty" yaml:"repository,omitempty"`
+}
+
+// SyntheticsAutomatedTestJobResult - Information on a job that was apart of a batch of automated test jobs
+type SyntheticsAutomatedTestJobResult struct {
+	// Test overrides
+	AutomatedTestMonitorConfig SyntheticsAutomatedTestMonitorConfig `json:"automatedTestMonitorConfig,omitempty"`
+	// Job batch Id
+	BatchId string `json:"batchId,omitempty"`
+	// Job duration
+	Duration int `json:"duration,omitempty"`
+	// Job error if any
+	Error string `json:"error,omitempty"`
+	// Job Id
+	ID string `json:"id,omitempty"`
+	// Job location
+	Location string `json:"location,omitempty"`
+	// Job location label
+	LocationLabel string `json:"locationLabel,omitempty"`
+	// Monitor nr1 entity guid
+	MonitorGUID EntityGUID `json:"monitorGuid,omitempty"`
+	// Job monitor Id
+	MonitorId string `json:"monitorId,omitempty"`
+	// Job monitor name
+	MonitorName string `json:"monitorName,omitempty"`
+	// Job result
+	Result SyntheticsJobStatus `json:"result,omitempty"`
+	// Link to job results
+	ResultsURL string `json:"resultsUrl,omitempty"`
+	// Job type
+	Type SyntheticsMonitorType `json:"type,omitempty"`
+	// Job type label
+	TypeLabel string `json:"typeLabel,omitempty"`
+}
+
+// SyntheticsAutomatedTestMonitorConfig - Monitor specific test config
+type SyntheticsAutomatedTestMonitorConfig struct {
+	// Specifies whether a failure of this monitor should fail the entire automated test
+	IsBlocking bool `json:"isBlocking,omitempty"`
+	// Specific overrides for the given monitor
+	Overrides SyntheticsAutomatedTestOverrides `json:"overrides,omitempty"`
+}
+
+// SyntheticsAutomatedTestMonitorConfigInput - Monitor specific test configuration
+type SyntheticsAutomatedTestMonitorConfigInput struct {
+	// Specifies whether a failure of this monitor should fail the entire automated test
+	IsBlocking bool `json:"isBlocking" yaml:"isBlocking"`
+	// Specific overrides for the given monitor
+	Overrides *SyntheticsAutomatedTestOverridesInput `json:"overrides" yaml:"overrides"`
```

Comment:
`omitEmpty` manually discarded from these attributes, because
- if `isBlocking` is given `false` in the YAML file, `isBlocking = false` is not even sent in the request to the API, which does not affect the response of the API, but would seem unexpected to anyone debugging this.
- if no `overrides` are specified in the YAML file, all contents of the `overrides` object are created as empty strings `""` which is sometimes leading to errors; hence, a pointer `*` has been added. However, with `omitEmpty`, if no `overrides` are specified in the input, it still disappears from the API request, so it has been removed.